pkg/auth/providers/common/ldap/ldap_util.go

```diff
@@ -44,6 +44,34 @@ func NewLDAPConn(config *v3.ActiveDirectoryConfig, caPool *x509.CertPool) (*ldap
 	return lConn, nil
 }
 
+func NewLDAPConnForOpenLDAP(config *v3.OpenLDAPConfig, caPool *x509.CertPool) (*ldapv2.Conn, error) {
+	logrus.Debug("Now creating Ldap connection")
+	var lConn *ldapv2.Conn
+	var err error
+	var tlsConfig *tls.Config
+	// TODO implment multi-server support
+	if len(config.Servers) != 1 {
+		return nil, errors.New("invalid server config. only exactly 1 server is currently supported")
+	}
+	server := config.Servers[0]
+	if config.TLS {
+		tlsConfig = &tls.Config{RootCAs: caPool, InsecureSkipVerify: false, ServerName: server}
+		lConn, err = ldapv2.DialTLS("tcp", fmt.Sprintf("%s:%d", server, config.Port), tlsConfig)
+		if err != nil {
+			return nil, fmt.Errorf("Error creating ssl connection: %v", err)
+		}
+	} else {
+		lConn, err = ldapv2.Dial("tcp", fmt.Sprintf("%s:%d", server, config.Port))
+		if err != nil {
+			return nil, fmt.Errorf("Error creating connection: %v", err)
+		}
+	}
+
+	lConn.SetTimeout(time.Duration(config.ConnectionTimeout) * time.Second)
+
+	return lConn, nil
+}
+
```

Comment:
I think only difference is you need to  pass the OpenLDAPConfig here instead of ActiveDirectoryConfig. The common logic should not be repeated, make a separate method to execute the common logic.

Reply:
You could change the signature to just take the specific fields you need off of the config. So the signature might look something like
NewLDAPConnForOpenLDAP(servers []string, tls bool, ... etc etc ..., caPool *x509.CertPool) (*ldapv2.Conn, error)